### ISSUE TITLE
Handle no-source and broken sources

### ISSUE THREAD

**Issue by heff** [videojs/video.js#869]
In the case where no supported sources are supplied to the video tag, the player should show a broken playback image.

View this example to see how different platforms handle the issue.
http://jsbin.com/OXEyuboL/1/edit

Case: no sources supplied at all (#764)
--> Empty video, no icon

Case: sources supplied, none supported or file error (#868)
--> Broken playback icon

In both cases:
- the play button should not show
- the spinner should not show

See #764 for more discussion around this.

Also see #796 and #798 for a beginning implementation. And #746.

Based on the [media-load-algorithm](http://www.w3.org/TR/html5/embedded-content-0.html#concept-media-load-algorithm):

```
IF there is no src attribute or source elements
--> Do not fire an error
--> Set networkState to NETWORK_EMPTY
--> Show disabled or no controls
        Safari Desktop: Controls, "loading...", play button responds to click but no result or error
        Safari iOS: No controls
        Chrome: Controls are grayed and disabled
        Firefox: Controls still work, including big play button, but nothing happens after clicking play
IF there is an empty src attribute or failure to load the src attribute source
--> Set networkState to NETWORK_NO_SOURCE
--> Fire an error event on the player (no bubble)
--> Set player.error() to 4 - MEDIA_ERR_SRC_NOT_SUPPORTED
--> Show disabled or no controls
        Firefox shows an X icon, "video format or mime type is not supported"
        Chrome shows disabled controls
        Safari Desktop shows controls that don't work. "loading..." is NOT displayed
IF there are source elements and each source element is not supported or fails
--> Set networkState to NETWORK_NO_SOURCE
--> Do not fire an error
--> Show disabled or no controls
        Firefox shows an X icon, "No video with supported format and MIME type found"
        Chrome shows disabled controls
        Safari Desktop shows controls that don't work. "loading..." is NOT displayed
```

It seems a little odd that when no source elements are supported, the player goes into a waiting state instead of firing an error. In order to know that the source selection algorithm failed on source elements you would have to listen for an error on the last source element. I don't think we could manually trigger that error on a source element, so that would make it hard to duplicate this process.

**Comment by ReNDeZOwnler** [comment]
I think we can handle the "no source" error in each technology.

Html5: on vjs.Html5.prototype.eventHandler
Flash: on vjs.Flash['onError']
Hls: on this.playlists.on('error', ...

If we detect the "no source" error, we can trigger a custom event ('srcnotfound') and retry the sources in a handler ('onSrcNotFound').

I have typed it with expected results. Yes, with a test would be better.

**Comment by mmcc** [comment]
Seems to be fixed in 4.12
